docs/tools/mbed_cli_2/migration_guide.md

```diff
@@ -0,0 +1,198 @@
+command line when compiling with `mbed-tools compile --toolchain GCC_ARM`, to,
+for example, select `GCC_ARM` as the toolchain.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli target` (set
+or get default target) is deprecated. Pass the desired target on the command
+line when compiling with `mbed-tools compile --mbed-target
+DISCO_L475VG_IOT01A`, to, for example, select `DISCO_L475VG_IOT01A` as the
+Mbed target.</span><br>
+
+
+### Creating an Mbed program
+<span style="background-color:#ffe6cc;">The subcommand `mbed-tools new` creates
+a new Mbed program by default and supports only one command-line-option
+`--create-only` (create a program without fetching mbed-os). The following
+command-line options, previously supported in Mbed CLI 1, are deprecated:
+
+* `--scm`: We now always use git.
+* `--program`: There is no need to specify this. We always assume a program.
+* `--library`: There is no need to specify this. We always assume a program.
+* `--mbedlib`: Importing Mbed 2 is no longer supported.
+* `--depth`: This is no longer needed, as Mbed CLI 2 automatically uses the
+  shallowest clone or fetch possible.
+* `--protocol`: This is now determined from the URL instead.
+* `--offline`: This is not needed, as there is no longer any Mbed-specific git
+  cache.
+* `--no-requirements`: Mbed CLI 2 never automatically installs requirements.
+</span><br>
+
+### Importing an Mbed program
+<span style="background-color:#ffe6cc;">The subcommand `mbed-tools
+import` clones an Mbed project and library dependencies. The following
+command-line-options supported in Mbed CLI 1 are deprecated:
+
+* `--depth`: The new CLI opportunistically imports the most shallow repository
+  possible. Running `git clone --depth <depth> <repo_url> | cd <repo_name> &&
+  mbed-tools deploy` will create an Mbed project with the required depth.
+* `--protocol`: To select the protocol, use a URL with the
+  desired protocol. For instance, you may import projects or libraries over SSH
+  (Secure Shell) as follows `mbed-tools import
+  git@github.com:ARMmbed/<some_driver>`, or with https `mbed-tools import
+  https://github.com/ARMmbed/<some_driver>`. Alternatively, if you would prefer
+  to always use SSH, even when your library or its dependencies specify https
+  in their `.lib` files, you can add the following into your .gitconfig to
+  force git to use SSH instead of https when fetching from GitHub.
+
+```sh
+[url "ssh://git@github.com/"]
+  insteadOf = https://github.com/
+```
+
+* `--offline`: The new tool doesn't maintain a git cache so we don't have the
+  option to use locally cached repositories.
+* `--no-requirements`
+* `--insecure`
+
+</span><br>
+
+### Configuring an Mbed program
+
+<span style="background-color:#ccffcc;">The subcommand `mbed-tools configure`
+generates an Mbed OS config CMake file named `mbed_config.cmake` in the path
+`cmake_build/<TARGET_NAME>/<PROFILE>/<TOOLCHAIN>/`.</span><br>
+
+### Compiling an Mbed program
+
```

Comment:
Another previously-undocumented Mbed CLI 1 feature is also not supported by Mbed CLI 2: use `detect` in place of a target name in `mbed-cli compile -m detect`, if only one board is connected. Is it something we want to drop?